src/sidebar/util/observe-element-size.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+/**
+ * Watch for changes in the size (`clientWidth` and `clientHeight`) of
+ * an element.
+ *
+ * Returns a cleanup function which should be called to remove observers when
+ * updates are no longer needed.
+ *
+ * @param {Element} element - HTML element to watch
+ * @param {(width: number, height: number) => any} onSizeChanged -
+ *   Callback to invoke with the `clientWidth` and `clientHeight` of the
+ *   element when a change in its size is detected.
+ * @return {() => void}
+ */
+function observeElementSize(element, onSizeChanged) {
+  if (typeof ResizeObserver !== 'undefined') {
+    const observer = new ResizeObserver(() =>
+      onSizeChanged(element.clientWidth, element.clientHeight)
+    );
+    observer.observe(element);
+    return () => observer.disconnect();
+  }
+
+  // Fallback method which listens for the most common events that result in
+  // element size changes:
+  //
+  // - Window size change
+  // - Media loading and adjusting size to content
+  // - DOM changes
+  //
+  // This is not comprehensive but it is simple to implement and good-enough for
+  // our current use cases.
+
+  let prevWidth = element.clientWidth;
+  let prevHeight = element.clientHeight;
+
+  const check = () => {
+    if (
+      prevWidth !== element.clientWidth ||
+      prevHeight !== element.clientHeight
+    ) {
+      prevWidth = element.clientWidth;
+      prevHeight = element.clientHeight;
+      onSizeChanged(prevWidth, prevHeight);
+    }
+  };
+
+  element.addEventListener('load', check);
```

Comment:
In this fallback variant with `MutationObserver`, you're also explicitly listening for `load` and `resize` events. Are these handled more automagically by `ResizeObserver`? I note that these events are tested for explicitly only on the "fallback" branch of the tests. Do we need tests for these events on the `ResizeObserver` branch as well, to make sure they're taken care of?

Reply:
`ResizeObserver` monitors when an element's size changes, regardless of cause. Listening for these DOM events is only necessary in the fallback case when we can't monitor element size changes directly but need to listen for other things which might result in size changes.